src/Microsoft.Data/PrimitiveDataFrameColumnArithmetic.cs

```diff
@@ -0,0 +1,4138 @@
+﻿
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Data
+{
+    internal interface IPrimitiveDataFrameColumnArithmetic<T>
+        where T : struct
+    {
+       void Add(PrimitiveDataFrameColumnContainer<T> left, PrimitiveDataFrameColumnContainer<T> right);
```

Comment:
Consider sorting. It's at least a stable ordering..

Reply:
Next patch. This'll cause too much pain when I rebase